components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -64,6 +65,13 @@ internal class EngineObserver(
         }
     }
 
+    private fun isUrlSame(originalUrl: String, newUrl: String): Boolean {
+        val originalUri = URI(originalUrl)
+        val uri = URI(newUrl)
```

Comment:
The constructor of [URI](https://developer.android.com/reference/java/net/URI.html#URI(java.lang.String)) can throw `URISyntaxException`. In other places we use [Uri.parse()](https://developer.android.com/reference/android/net/Uri.html?hl=en#parse(java.lang.String)) which is more forgiving.